plugins/modules/panos_state_snapshot.py

```diff
@@ -0,0 +1,152 @@
+notes:
+    - Only Firewalls are supported.
+    - Check mode is not supported.
+extends_documentation_fragment:
+    - paloaltonetworks.panos.fragments.transitional_provider
+    - paloaltonetworks.panos.fragments.vsys
+options:
+    state_areas:
+        description:
+            - A list of Firewall state areas that we should take a snapshot of. For the details on currently supported list please refer to
+              L(package documentation, https://pan.dev/panos/docs/panos-upgrade-assurance/configuration-details/#state-snapshots).
+            - To capture the actual snapshot data use a register.
+        type: list
+        elements: str
+        default: ["all"]
+"""
+
+EXAMPLES = """
+- name: Run snapshot of all areas except for session statistics
+  panos_state_snapshot:
+    provider: '{{ device }}'
+    state_areas:
+      - '!session_stats'
+    register: snapshot
+"""
+
+RETURN = """
+response:
+    description:
+        - This is a B(dict) where keys are state areas names just as you specify them in the I(state_areas) property.
+        - Values contain the snapshot data. Type and structure differs per state area. Please refer to
+          L(package documentation, https://pan.dev/panos/docs/panos-upgrade-assurance/configuration-details/#state-snapshots) for details.
+    type: dict
+    returned: always
+    sample:
+        arp_table: {}
+        content_version:
+            version: 8635-7675
+        ip_sec_tunnels: {}
+        license:
+            DNS Security:
+                authcode: null
+                base-license-name: PA-VM
+                description: Palo Alto Networks DNS Security License
+                expired: no
+                expires: December 31, 2023
+                feature: DNS Security
+                issued: April 13, 2023
+                serial: "xxxxxxxxxxxxxxxxx"
+            PA-VM:
+                authcode: null
+                description: Standard VM-300
+                expired: no
+                expires: December 31, 2023
+                feature: PA-VM
+                issued: April 13, 2023
+                serial: "xxxxxxxxxxxxxxxxx"
+        routes: {}
+        session_stats:
+            age-accel-thresh: "80"
+            age-accel-tsf: "2"
+            age-scan-ssf: "8"
+            age-scan-thresh: "80"
+            age-scan-tmo: "10"
+            cps: "0"
+            dis-def: "60"
+            dis-sctp: "30"
+            dis-tcp: "90"
+            dis-udp: "60"
+            icmp-unreachable-rate: "200"
+            kbps: "0"
+            max-pending-mcast: "0"
+            num-active: "0"
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.paloaltonetworks.panos.plugins.module_utils.panos import (
+    get_connection,
+)
+
+MIN_PUA_VER = (0, 3, 0)
+
+try:
+    from panos_upgrade_assurance.firewall_proxy import FirewallProxy
+    from panos_upgrade_assurance.check_firewall import CheckFirewall
+except ImportError:
+    pass
```

Comment:
Why don't we have the `PUA_AVAILABLE` logic here as in the `panos_snapshot_report.py`?

Reply:
no, it's in the helper class, panos_snapshot_report does not use the helper class, so the logic had to be moved to the module directly